hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -134,26 +150,36 @@ public void doZeroSum(final List<BalanceChange> changes) {
             }
         }
 
+        if (validity == OK
+                && (autoCreationFee > 0)
+                && (autoCreationFee > (long) accountsLedger.get(txnCtx.activePayer(), BALANCE))) {
+            validity = INSUFFICIENT_PAYER_BALANCE;
+        }
+
         if (validity == OK) {
             adjustBalancesAndAllowances(changes);
             if (autoCreationFee > 0) {
-                payFunding(autoCreationFee);
+                payAutoCreationFee(autoCreationFee);
                 autoCreationLogic.submitRecordsTo(recordsHistorian);
+                autoCreationLogic.clearTokenAliasMap();
```

Comment:
I think we need to call `tokenAliasMap.clear()` in `AutoCreationLogic.reset()`, since that is guaranteed to be called at the beginning of every `HederaLedger` transaction. (And then we don't need to call it here.)

Reply:
Yeah , but I am lazily initiating the HashMap only if there is tokenTransfers to alias. I think it should be fine to just initialize like `pendingCreations` , since it will be reused anyways.

Reply:
Initialized map irrespective of token transfers to alias and did as suggested.